channel-sdk/src/main/java/com/zello/channel/sdk/image/ImageMessageManager.kt

```diff
@@ -58,15 +59,18 @@ internal data class IncomingImageInfo(val imageId: Int, val sender: String, val
 		private set
 }
 
-// TODO: Replace GlobalScope with a class-specific coroutine scope (Global scope is recommended against?)
 internal class ImageMessageManagerImpl(private val listener: ImageMessageManagerListener,
-									   private val backgroundScope: CoroutineScope = GlobalScope,
+									   private val backgroundScope: CoroutineScope = CoroutineScope(Dispatchers.Default),
 									   private val mainThreadDispatcher: CoroutineDispatcher = Dispatchers.Main,
 									   private val incomingImageTimeout: Long = 2 * 60 * 1000L) : ImageMessageManager {
 
 	private var incomingImages = hashMapOf<Int, IncomingImageInfo>()
 
-	override fun sendImage(image: Bitmap, transport: Transport, recipient: String?, continuation: SentImageCallback?) {
+	override fun isTypeValid(type: String): Boolean {
+		return "jpeg" == type
+	}
+
+	override fun sendImage(image: Bitmap, transport: Transport, recipient: String?, continuation: (Int, SendImageError?) -> Unit) {
```

Comment:
I like that you've replaced the `SentImageCallback` interface with a lambda 👍. Curious about the choice to change it from optional to non-optional: did you originally anticipate some use case where it would be null, but changed your mind about the validity of the use case? Or what happened there?

Reply:
I thought it would be part of the public API, then realized while I was cleaning up that we already have a way to tell the user about failed messages (`SessionListener.onError()`) and they probably don't care about the image id. So I made it internal and non-optional. Changed to a lambda for the same reason: The interface makes it easier to call from Java, but there are no Java customers for the internal API.